ddl/ddl.go

```diff
@@ -1099,6 +1354,82 @@ func CancelJobs(txn kv.Transaction, ids []int64) ([]error, error) {
 	return errs, nil
 }
 
+// cancelConcurrencyJobs cancels the DDL jobs that are in the concurrent state.
+func cancelConcurrencyJobs(se sessionctx.Context, ids []int64) ([]error, error) {
+	failpoint.Inject("mockCancelConcurencyDDL", func(val failpoint.Value) {
+		if val.(bool) {
+			failpoint.Return(nil, errors.New("mock commit error"))
+		}
+	})
+	if len(ids) == 0 {
+		return nil, nil
+	}
+	var jobSet = make(map[int64]int) // jobID -> error index
+
+	sess := newSession(se)
+	err := sess.begin()
+	if err != nil {
+		return nil, err
+	}
+
+	idsStr := make([]string, 0, len(ids))
+	for idx, id := range ids {
+		jobSet[id] = idx
+		idsStr = append(idsStr, strconv.FormatInt(id, 10))
+	}
+
+	jobs, err := getJobsBySQL(sess, JobTable, fmt.Sprintf("job_id in (%s) order by job_id", strings.Join(idsStr, ", ")))
+	if err != nil {
+		sess.rollback()
+		return nil, err
+	}
+
+	errs := make([]error, len(ids))
+
+	for _, job := range jobs {
+		i, ok := jobSet[job.ID]
+		if !ok {
+			continue
+		}
+		delete(jobSet, job.ID)
+		if job.IsDone() || job.IsSynced() {
+			errs[i] = dbterror.ErrCancelFinishedDDLJob.GenWithStackByArgs(job.ID)
+			continue
+		}
+		// If the state is rolling back, it means the work is cleaning the data after cancelling the job.
```

Comment:
Does this comment need to move after [line1403](https://github.com/pingcap/tidb/pull/32169/files#diff-a185df836b8926c4afe351d4def1b56d1e282c65aabe6677c2fa2c54590cc4b2R1403)?